### PR TITLE
Add NATS client to the Network Programming section

### PR DESCRIPTION
**Pull request by stjepang**
The `nats` crate is the official client for the [NATS](https://en.wikipedia.org/wiki/Cloud_Native_Computing_Foundation#NATS) messaging system.

https://nats.io

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1358,6 +1358,8 @@ See also [Are we game yet?](https://arewegameyet.rs)
   * [mozilla/neqo](https://github.com/mozilla/neqo) — an Implementation of QUIC written in Rust
 * P2P
   * [libp2p/rust-libp2p](https://github.com/libp2p/rust-libp2p) — The Rust Implementation of libp2p networking stack. [![Circle CI](https://circleci.com/gh/libp2p/rust-libp2p.svg?style=svg)](https://circleci.com/gh/libp2p/rust-libp2p)
+* NATS
+  * [nats-io/nats.rs](https://github.com/nats-io/nats.rs) — Rust client for NATS, the cloud native messaging system.
```
Comment (palfrey) [inline README.md:1362]: I'm seeing a Github Actions build there. Can you please add the build badge to this as per https://github.com/rust-unofficial/awesome-rust/blob/master/CONTRIBUTING.md ?

Comment (stjepang) [reply README.md]: Done.

**Comment by stjepang** [comment]
CI seems to be failing because the master branch is not specified in the link to the badge.

I don't understand why this link works:

https://github.com/nats-io/nats.rs/workflows/Rust/badge.svg

But this one doesn't:

https://github.com/nats-io/nats.rs/workflows/Rust/badge.svg?branch=master

Am I doing something wrong?

**Comment by palfrey** [comment]
> CI seems to be failing because the master branch is not specified in the link to the badge.
> 
> I don't understand why this link works:
> 
> https://github.com/nats-io/nats.rs/workflows/Rust/badge.svg
> 
> But this one shows "no status":
> 
> https://github.com/nats-io/nats.rs/workflows/Rust/badge.svg?branch=master
> 
> Am I doing something wrong?

So, that's actually the right answer. https://github.com/nats-io/nats.rs/actions doesn't contain any builds for master!

The logic in the workflow file of 
```
on:
  pull_request:
    branches:
    - master
```
says just to run builds on pull requests that merge to master. If you want master builds as well, you'll want `push` events filtered down to master.

**Pull request merged**